### PR TITLE
Decompile overlay_50

### PR DESCRIPTION
**Pull request by abaresk**

### CODE REVIEW

File: include/overlay_50.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    u8 unk4b:1;
+    u8 unk5[0x63];
+    u8 unk68[0x600];
+    u8 unk668[6];
+    SAVEDATA *saveData;
+    u32 unk674;
+} Unk021E5A3C;
+
+typedef struct {
+    u32 heap_id;
```
[the remark this answers is not in the document]
Comment (abaresk) [reply include/overlay_50.h]: Done

File: include/overlay_50.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+typedef struct {
+    u8 unk0[4];
+    u8 unk4a:4;
+    u8 unk4b:1;
+    u8 unk5[0x63];
+    u8 unk68[0x600];
+    u8 unk668[6];
+    SAVEDATA *saveData;
+    u32 unk674;
+} Unk021E5A3C;
```
Comment (PikalaxALT) [inline include/overlay_50.h:15]: If you're using hex addresses for structs defined in overlay modules, you should also specify which overlay they are tied to. Reduces the chance of namespace collisions.

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: include/overlay_50.h (hunk 2 above)
Comment (abaresk) [reply include/overlay_50.h]: Done

File: include/overlay_50.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    HeapID heap_id;
+    u8 unk4[4];
+    OVY_MANAGER *ov_mgr;
+    Ov50_021E5A3C *unkC;
+    void *unk10;
+} Ov50_021E5924;
+
+BOOL ov50_021E5924(OVY_MANAGER *man);
+BOOL ov50_021E595C(OVY_MANAGER *man, int *state);
+BOOL ov50_021E59A8(OVY_MANAGER *man);
```
Comment (tgsm) [inline include/overlay_50.h:27]: All of these take in an int*
```suggestion
BOOL ov50_021E5924(OVY_MANAGER *man, int *state);
BOOL ov50_021E595C(OVY_MANAGER *man, int *state);
BOOL ov50_021E59A8(OVY_MANAGER *man, int *state);
```

You may need to add `#pragma unused(state)` to the first line of their definitions.

**Review by tgsm — CHANGES_REQUESTED** [review]
Looks fine to me, just one nit

File: include/overlay_50.h (hunk 3 above)
Comment (abaresk) [reply include/overlay_50.h]: Done

**Review by tgsm — APPROVED** [review]

**Pull request merged**